src/local_scheduler/local_scheduler.cc

```diff
@@ -342,42 +344,29 @@ LocalSchedulerState *LocalSchedulerState_init(
 
   /* Connect to Redis if a Redis address is provided. */
   if (redis_primary_addr != NULL) {
-    int num_args;
-    const char **db_connect_args = NULL;
-    /* Use UT_string to convert the resource value into a string. */
-    UT_string *num_cpus;
-    UT_string *num_gpus;
-    utstring_new(num_cpus);
-    utstring_new(num_gpus);
-    utstring_printf(num_cpus, "%f", static_resource_conf[0]);
-    utstring_printf(num_gpus, "%f", static_resource_conf[1]);
+    /* Use std::string to convert the resource value into a string. */
+    std::string num_cpus = std::to_string(static_resource_conf[0]);
+    std::string num_gpus = std::to_string(static_resource_conf[1]);
+
+    /* Construct db_connect_args */
+    int num_args = 6;
+    std::vector<const char *> db_connect_args;
```

Comment:
I would've preferred to type `db_connect_args` as `std::vector<std::string>`. Unfortunately, this isn't easily possible without modifying the definition of `db_connect()`.

It might be valuable to to redefine `db_connect` to use accept `std::string` and `std::vector<std::string>` args to clean up the code.